sigopt/ratelimit.py

```diff
@@ -0,0 +1,27 @@
+import random
+import threading
+import time
+
+
+global_failed_status_count = 0
+thread_lock = threading.Lock()
+
+class FailedStatusRateLimit(object):
+  def __init__(self, limit):
+    self.limit = limit
+
+  def increment_and_check(self):
+    with thread_lock:
+      global global_failed_status_count
+      global_failed_status_count += 1
+
+    multiples_over = global_failed_status_count // self.limit
```

Comment:
probably want to do this calculation while you've acquired the thread

Reply:
Wasn't sure if it mattered, but probably best practice to be clear, so yeah. Updated

Reply:
you could increment the `global_failed_status_count`, another thread could call `.clear()` then `global_failed_status_count` would be 0 here (when it maybe should have triggered a backoff)

Reply:
if another thread cleared it (with a successful call, presumably), then we don't need to make them backoff any longer, so it should be fine. But i see, you're recommending do it in the same check as the increment, that makes sense

Reply:
fixed